### PR TITLE
Added kustomize deployment with cert-manager

### PR DESCRIPTION
**Pull request by jasonrichardsmith**
*Issue #, if available:*
Deploy without cert requirements #3 

*Description of changes:*
Added kustomize files and instructions on how to deploy with cert-manager.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by bcelenza** [comment]
Hey Jason, can you help me understand how cert-manager works in the context of Envoy and App Mesh? I'm working on encryption for App Mesh currently, and want to make sure I'm thinking through all the variants. I read the documentation for cert-manager but it's not obvious to me how certificates are actually bound (or where they are bound).

**Comment by nckturner** [comment]
Let's also think through our installation methods.  Does this belong as kustomize here, or should we add it to the new injector helm chart? https://github.com/aws/eks-charts/tree/master/stable/appmesh-inject

**Comment by jasonrichardsmith** [comment]
@nckturner why not both?

**Comment by stefanprodan** [comment]
I think this should replace the bash scripts. So in the end there will be two install options: one with Helm(no additional TLS controllers required) and one will kubectl (cert-manager required). I also think that both Helm and Kustomize installers should deploy the App Mesh controllers in a single namespace (appmesh-system) and the deployments/services should have a consistent naming like: appmesh-inject and appmesh-controller. Why drop the install scripts? For once, the scripts are not portable, they don’t work on Windows and they depend on openssl and other packages on the host os.

**Comment by nckturner** [comment]
Yeah, I want to limit confusion for users, and I think dropping the bash
scripts is a good way to do that, as long as there is a fine alternative
non-helm users.  In terms of using the same namespace for both and
consistent naming, I agree.

File: INSTALL.md [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+kind: Kustomization
+bases:
+- github.com/aws/aws-app-mesh-inject
+patches:
+- mymesh.yaml
+```
+
+mymesh.yaml
+```yaml
+
```
[the remark this answers is not in the document]
Comment (jasonrichardsmith) [reply INSTALL.md]: fixed

File: INSTALL.md [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+mkdir myappmeshconfig
+```
+Create two files in the folder:
+
+kustomization.yaml
+```yaml
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+bases:
+- github.com/aws/aws-app-mesh-inject
```
Comment (stefanprodan) [inline INSTALL.md]:
```suggestion
- github.com/aws/aws-app-mesh-inject//kustomize
```

File: INSTALL.md [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+```
+
+mymesh.yaml
+```yaml
+---
+apiVersion: apps/v1beta1
+kind: Deployment
+metadata:
+  name: aws-app-mesh-inject
+  namespace: appmesh-inject
```
Comment (stefanprodan) [inline INSTALL.md]:
```suggestion
  namespace: appmesh-system
```

File: INSTALL.md [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+- mymesh.yaml
+```
+
+mymesh.yaml
+```yaml
+---
+apiVersion: apps/v1beta1
+kind: Deployment
+metadata:
+  name: aws-app-mesh-inject
```
Comment (stefanprodan) [inline INSTALL.md]:
```suggestion
  name: appmesh-inject
```

File: kustomize/certs.yaml [hunk 5]
```diff
@@ -0,0 +1,26 @@
+apiVersion: certmanager.k8s.io/v1alpha1
+kind: Issuer
+metadata:
+  name: selfsigned
+  namespace: appmesh-inject
```
Comment (stefanprodan) [inline kustomize/certs.yaml]:
```suggestion
  namespace: appmesh-system
```

File: kustomize/certs.yaml [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+spec:
+  selfSigned: {}
+---
+apiVersion: certmanager.k8s.io/v1alpha1
+kind: Certificate
+metadata:
+  name: inject-cert
+  namespace: appmesh-inject
```
Comment (stefanprodan) [inline kustomize/certs.yaml]:
```suggestion
  namespace: appmesh-system
```

File: kustomize/certs.yaml [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+spec:
+  isCA: false
+  dnsNames:
+    - aws-app-mesh-inject
```
Comment (stefanprodan) [inline kustomize/certs.yaml]:
```suggestion
    - appmesh-inject
```

File: kustomize/certs.yaml [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+    - aws-app-mesh-inject.appmesh-inject
```
Comment (stefanprodan) [inline kustomize/certs.yaml]:
```suggestion
    - appmesh-inject.appmesh-system
```

File: kustomize/certs.yaml [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+    - aws-app-mesh-inject.appmesh-inject.svc
```
Comment (stefanprodan) [inline kustomize/certs.yaml]:
```suggestion
    - appmesh-inject.appmesh-system.svc
```

File: kustomize/cluster-roles.yaml [hunk 10]
```diff
@@ -0,0 +1,29 @@
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: aws-app-mesh-inject-sa
```
Comment (stefanprodan) [inline kustomize/cluster-roles.yaml]:
```suggestion
  name: appmesh-inject
```

File: kustomize/cluster-roles.yaml [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+  namespace: appmesh-inject
```
Comment (stefanprodan) [inline kustomize/cluster-roles.yaml]:
```suggestion
  namespace: appmesh-system
```

File: kustomize/cluster-roles.yaml [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+---
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1beta1
+metadata:
+  name: aws-app-mesh-inject-cr
```
Comment (stefanprodan) [inline kustomize/cluster-roles.yaml]:
```suggestion
  name: appmesh-inject
```

File: kustomize/cluster-roles.yaml [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+    resources: ["replicasets"]
+    verbs: ["get"]
+---
+kind: ClusterRoleBinding
+apiVersion: rbac.authorization.k8s.io/v1beta1
+metadata:
+  name: binding
+subjects:
+  - kind: ServiceAccount
+    name: aws-app-mesh-inject-sa
```
Comment (stefanprodan) [inline kustomize/cluster-roles.yaml]:
```suggestion
    name: appmesh-inject
```

File: kustomize/cluster-roles.yaml [hunk 14]
```diff
@@ -0,0 +1,29 @@
...
+    namespace: appmesh-inject
```
Comment (stefanprodan) [inline kustomize/cluster-roles.yaml]:
```suggestion
    namespace: appmesh-system
```

File: kustomize/cluster-roles.yaml [hunk 15]
```diff
@@ -0,0 +1,29 @@
...
+    apiGroup: ""
+roleRef:
+  kind: ClusterRole
+  name: aws-app-mesh-inject-cr
```
Comment (stefanprodan) [inline kustomize/cluster-roles.yaml]:
```suggestion
  name: appmesh-inject
```

File: kustomize/deployment.yaml [hunk 16]
```diff
@@ -0,0 +1,71 @@
+---
+apiVersion: apps/v1beta1
+kind: Deployment
+metadata:
+  name: aws-app-mesh-inject
```
Comment (stefanprodan) [inline kustomize/deployment.yaml]:
```suggestion
  name: appmesh-inject
```

File: kustomize/deployment.yaml [hunk 17]
```diff
@@ -0,0 +1,71 @@
...
+  namespace: appmesh-inject
```
Comment (stefanprodan) [inline kustomize/deployment.yaml]:
```suggestion
  namespace: appmesh-system
```

File: kustomize/deployment.yaml [hunk 18]
```diff
@@ -0,0 +1,71 @@
...
+  labels:
+    name: aws-app-mesh-inject
```
Comment (stefanprodan) [inline kustomize/deployment.yaml]:
```suggestion
    name: appmesh-inject
```

File: kustomize/deployment.yaml [hunk 19]
```diff
@@ -0,0 +1,71 @@
...
+spec:
+  replicas: 1
+  template:
+    metadata:
+      name: aws-app-mesh-inject
```
Comment (stefanprodan) [inline kustomize/deployment.yaml]:
```suggestion
      name: appmesh-inject
```

File: kustomize/deployment.yaml [hunk 20]
```diff
@@ -0,0 +1,71 @@
...
+      labels:
+        name: aws-app-mesh-inject
```
Comment (stefanprodan) [inline kustomize/deployment.yaml]:
```suggestion
        name: appmesh-inject
```

File: kustomize/deployment.yaml [hunk 21]
```diff
@@ -0,0 +1,71 @@
...
+    spec:
+      serviceAccountName: aws-app-mesh-inject-sa
```
Comment (stefanprodan) [inline kustomize/deployment.yaml]:
```suggestion
      serviceAccountName: appmesh-inject
```

File: kustomize/ns.yaml [hunk 22]
```diff
@@ -0,0 +1,4 @@
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: appmesh-inject
```
Comment (stefanprodan) [inline kustomize/ns.yaml]:
```suggestion
  name: appmesh-system
```

File: kustomize/service.yaml [hunk 23]
```diff
@@ -0,0 +1,15 @@
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: aws-app-mesh-inject
```
Comment (stefanprodan) [inline kustomize/service.yaml]:
```suggestion
  name: appmesh-inject
```

File: kustomize/service.yaml [hunk 24]
```diff
@@ -0,0 +1,15 @@
...
+  namespace: appmesh-inject
```
Comment (stefanprodan) [inline kustomize/service.yaml]:
```suggestion
  namespace: appmesh-system
```

File: kustomize/service.yaml [hunk 25]
```diff
@@ -0,0 +1,15 @@
...
+  labels:
+    name: aws-app-mesh-inject
```
Comment (stefanprodan) [inline kustomize/service.yaml]:
```suggestion
    name: appmesh-inject
```

File: kustomize/service.yaml [hunk 26]
```diff
@@ -0,0 +1,15 @@
...
+spec:
+  ports:
+  - name: webhook
+    port: 443
+    targetPort: 8080
+  selector:
+    name: aws-app-mesh-inject
```
Comment (stefanprodan) [inline kustomize/service.yaml]:
```suggestion
    name: appmesh-inject
```

File: kustomize/webhookconfig.yaml [hunk 27]
```diff
@@ -0,0 +1,23 @@
+---
+apiVersion: admissionregistration.k8s.io/v1beta1
+kind: MutatingWebhookConfiguration
+metadata:
+  name: aws-app-mesh-inject
```
Comment (stefanprodan) [inline kustomize/webhookconfig.yaml]:
```suggestion
  name: appmesh-inject
```

File: kustomize/webhookconfig.yaml [hunk 28]
```diff
@@ -0,0 +1,23 @@
...
+  annotations:
+    certmanager.k8s.io/inject-ca-from: appmesh-inject/inject-cert
+webhooks:
+  - name: aws-app-mesh-inject.aws.amazon.com
```
Comment (stefanprodan) [inline kustomize/webhookconfig.yaml]:
```suggestion
  - name: appmesh-inject.aws.amazon.com
```

File: kustomize/webhookconfig.yaml [hunk 29]
```diff
@@ -0,0 +1,23 @@
...
+    clientConfig:
+      service:
+        name: aws-app-mesh-inject
```
Comment (stefanprodan) [inline kustomize/webhookconfig.yaml]:
```suggestion
        name: appmesh-inject
```

File: kustomize/webhookconfig.yaml [hunk 30]
```diff
@@ -0,0 +1,23 @@
...
+        namespace: appmesh-inject
```
Comment (stefanprodan) [inline kustomize/webhookconfig.yaml]:
```suggestion
        namespace: appmesh-system
```

**Comment by stefanprodan** [comment]
Please see my first suggestion, the base should end in //kustomize otherwise it will download the whole repo.

**Pull request merged**